core/utils.py

```diff
@@ -42,7 +46,7 @@ def centroid_normalization(dir, poi, X, alpha=0.7):
     dir = (alpha * dir * centroid_dist) / np.sqrt(dir@dir)
```

Comment:
Please consider renaming `dir` to direction, here and in the functions below operating on `dir`. 

https://google.github.io/styleguide/pyguide.html#316-naming

Reply:
This goes for the other usage in this file.

Reply:
We can revisit this when we push for open source quality. Both core/utils.py and visualize/two_d_plots.py have low-priority TODOs requesting refactor.